### PR TITLE
Refactor kubernetes provider by separating Loader and Applier

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

This separates Loader and Applier completely before updating the Applier to use the specified kube-config path.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nghialv** [comment]
There is a bug in our CodeGen job so please ignore it in this case.

File: pkg/app/piped/cloudprovider/kubernetes/loader.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+const (
+	TemplatingMethodHelm      TemplatingMethod = "helm"
+	TemplatingMethodKustomize TemplatingMethod = "kustomize"
+	TemplatingMethodNone      TemplatingMethod = "none"
+)
+
+type ManifestLoader interface {
+	// LoadManifests renders and loads all manifests for application.
+	LoadManifests(ctx context.Context) ([]Manifest, error)
+}
```
Comment (knanao) [inline pkg/app/piped/cloudprovider/kubernetes/loader.go:42]: nits, I would prefer either fixing to `ManifestLoader` into `Loader` or `LoadManifests` into `Load`.

Comment (nghialv) [reply pkg/app/piped/cloudprovider/kubernetes/loader.go:42]: Nice idea. Let me fix the interface name to `Loader`. 👍

Comment (nghialv) [reply pkg/app/piped/cloudprovider/kubernetes/loader.go]: Addressed by https://github.com/pipe-cd/pipecd/pull/3407/commits/f1d6831f41d9d0ae083d649e7aa8d0ebc52962aa

**Review by knanao — COMMENTED** [review]
Nice refactoring XD
/lgtm

**Comment by khanhtc1202** [comment]
Here you go 👌 
/approve

**Pull request merged**